pkg/services/store/k8saccess/service.go

```diff
@@ -0,0 +1,62 @@
+package k8saccess
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/grafana/grafana/pkg/api/routing"
+	"github.com/grafana/grafana/pkg/services/featuremgmt"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+type K8SAccess interface {
+	// Get the system client
+	GetSystemClient() *kubernetes.Clientset
+}
+
+var _ K8SAccess = &k8sAccess{}
+
+type k8sAccess struct {
+	apihelper *httpHelper
+	sysclient *kubernetes.Clientset
+	syserr    error
+}
+
+func ProvideK8SAccess(toggles featuremgmt.FeatureToggles, router routing.RouteRegister) K8SAccess {
+	access := &k8sAccess{}
+
+	if !toggles.IsEnabled(featuremgmt.FlagK8s) {
```

Comment:
Might be good to throw a comment here that explains the routes don't get mounted. I don't think that's obvious